### PR TITLE
Update Dio 5.0.0

### PR DESCRIPTION
**Pull request by SebastianKlaiber**
# Pull Request Template

## Your checklist for this pull request

Please review the [guidelines for contributing](https://github.com/lomsa-dev/http-mock-adapter/blob/develop/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a `feature`/`bugfix` branch** (right side). Don't request your `main` or `develop`!
- [x] Make sure you are making a pull request against the **`develop` branch** (left side). Also, you should start *your branch* off *our `develop`*.
- [x] Make sure that you add reviewers, assignees, labels, projects, milestones and linkes issues to the pull request as you see fit.
- [x] Check the commit's or even all commits' message styles to match our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

**Tip**: Delete everything but the following section that contains the description of the pull request.

---

## Description

Update packages to the latest versions.
Fix lint warnings and errors.

### CODE REVIEW

**Comment by chunghha** [comment]
@SebastianKlaiber  Nice! And thanks for this PR. I was doing same so left few tiny comments.

File: pubspec.yaml [hunk 1]
```diff
@@ -9,11 +9,11 @@ environment:
   sdk: ">=2.12.0 <3.0.0"
 
 dependencies:
-  collection: ^1.15.0
-  dio: ^4.0.0
-  http_parser: ^4.0.0
+  collection: ^1.17.1
```
Comment (kuhnroyal) [inline pubspec.yaml:12]: This bump isn't required and it prevents running tests on older SDKs, please revert to `collection: ^1.15.0`.

Comment (kuhnroyal) [reply pubspec.yaml:12]: This effectively requires Dart 2.18.0+ at the moment.

Comment (SebastianKlaiber) [reply pubspec.yaml]: The change is reverted.

**Comment by SebastianKlaiber** [comment]
> @SebastianKlaiber Nice! And thanks for this PR.
> 
> I was doing same so left few "tiny" comments which you can just safely ignore if you wouldn't mind to update.

Unfortunately, I can not see your comments.

**Comment by chunghha** [comment]
I aslo see my comment only if I am loggin in. Thus, here are recao:

1. sdk: ">=2.12.0 <3.0.0" <-- we may need to bump up this minimum version with 2.15.0 as the dio v5 does.
2.  as you removed the imports, this ignore comments can go away in 2 places --> // ignore: implementation_imports
3.  I also mentioned about I saw version conflict between collection and test so I set the cokkection version with ^1.17.0 but not tested with lowering to ^1.15.0 as @kuhnroyal  pointed.

**Comment by kuhnroyal** [comment]
> I aslo see my comment only if I am loggin in. 

You probably have to submit your PR review.

File: lib/src/matchers/matchers.dart [hunk 2]
```diff
@@ -1,6 +1,5 @@
 import 'package:dio/dio.dart';
 // ignore: implementation_imports
```
[the remark this answers is not in the document]
Comment (SebastianKlaiber) [reply lib/src/matchers/matchers.dart]: removed

File: lib/src/matchers/list_param.dart [hunk 3]
```diff
@@ -1,7 +1,6 @@
 import 'package:collection/collection.dart';
 import 'package:dio/dio.dart';
 // ignore: implementation_imports
```
[the remark this answers is not in the document]
Comment (SebastianKlaiber) [reply lib/src/matchers/list_param.dart]: removed

**Review by kuhnroyal — APPROVED** [review]

**Comment by SebastianKlaiber** [comment]
@chunghha @kuhnroyal do you know who can merge the PR?

**Comment by kuhnroyal** [comment]
@LukaGiorgadze @theiskaa Hello guys, can you potentially review/merge/release?

**Pull request merged**